cc/sgf.cc

```diff
@@ -191,6 +191,39 @@ class Parser {
   std::string* error_;
 };
 
+void GetMoveTreeImpl(const Ast::Tree& tree,
+                     std::vector<std::unique_ptr<MoveTree>>* dst) {
```

Comment:
I don't think that would help. The SGF syntax is a bit weird for a format that describes a tree since most often the game trees have a very low branching factor (~1 in most cases).

A list of nodes in the SGF source corresponds to a chain of descendants not siblings, so the shape of the source SGF tree doesn't actually correspond to the shape of the parsed game tree.

To give an example, a source tree that looks like this:
```
   (a b c)
   /     \
(d e)   (f g)
```

Parses to a game tree that looks like this:
```
(a)
 |
(b)
 |
(c)
 |  \
(d) (f)
 |   |
(e) (g)
```

Reply:
I misunderstood what was happening.